src/openfl/text/TextField.hx

```diff
@@ -3109,6 +3184,68 @@ class TextField extends InteractiveObject
 		return __textEngine.wordWrap = value;
 	}
 
+	@:noCompletion private function get_textDirection():TextDirection
+	{
+		if (__textDirection != INVALID) return __textDirection;
+		return __textEngine.mainDirection();
+	}
+
+	@:noCompletion private function set_textDirection(value:TextDirection):TextDirection
+	{
+		if (value != __textDirection)
+		{
+			__setTransformDirty();
+			__dirty = true;
+			__layoutDirty = true;
+			__setRenderDirty();
+		}
+
+		return __textDirection = value;
+	}
+
+	@:noCompletion private function get_script():Null<TextScript>
+	{
+		return __textEngine.script;
+	}
+
+	@:noCompletion private function set_script(value:Null<TextScript>):Null<TextScript>
+	{
+		if (value != __textEngine.script)
+		{
+			__setTransformDirty();
+			__dirty = true;
+			__layoutDirty = true;
+			__setRenderDirty();
+		}
+
+		return __textEngine.script = value;
+	}
+
+	@:noCompletion private function get_language():String
+	{
+		return __textEngine.language;
+	}
+
+
+	@:access(openfl.globalization.LocaleID.RTL_LANGUAGES)
+	@:noCompletion private function set_language(value:String):String
+	{
+		if (value != __textEngine.language)
+		{
+			__setTransformDirty();
+			__dirty = true;
+			__layoutDirty = true;
+			__setRenderDirty();
+		}
+
+		if (LocaleID.RTL_LANGUAGES.contains(value))
```

Comment:
This "contains" method call is causing compilation failures on older versions of Haxe.